### PR TITLE
Added ability to pass "remote id" to partner creation and account creation methods

### PR DESCRIPTION
**Pull request by ekoval**
Part of: https://github.com/yola/production/issues/3533

### CODE REVIEW

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: sitewit/services.py [hunk 1]
```diff
@@ -35,17 +35,24 @@ def _get_account_auth_header(self, account_token):
         return self._compose_auth_header((
             self._partner_id, self._partner_token, account_token))
 
-    def _get_partner_auth_header(self, subpartner_id=None):
+    def _get_partner_auth_headers(self, subpartner_id=None, remote_id=None):
```
Comment (RayeN) [inline sitewit/services.py]: What do you think about naming it `remote_subpartner_id` to be consistent with other params

File: sitewit/services.py [hunk 2]
```diff
@@ -35,17 +35,24 @@ def _get_account_auth_header(self, account_token):
...
+        headers = {}
         auth_list = [self._partner_id, self._partner_token]
+
         if subpartner_id is not None:
             auth_list.append(subpartner_id)
-        return self._compose_auth_header(auth_list)
+        elif remote_id is not None:
```
Comment (RayeN) [inline sitewit/services.py]: Aren't subpartner_id and remote_subpartner_id options mutually exclusive?

File: sitewit/services.py [hunk 3]
```diff
@@ -35,17 +35,24 @@ def _get_account_auth_header(self, account_token):
...
+        headers.update(self._compose_auth_header(auth_list))
+        return headers
 
     def _compose_auth_header(self, elements):
         return {'PartnerAuth': base64.b64encode(':'.join(elements))}
 
     def create_account(self, site_id, url, user_name, user_email,
-                       currency, country_code, user_token=None):
+                       currency, country_code, user_token=None,
+                       remote_id=None):
```
Comment (RayeN) [inline sitewit/services.py]: I would name it `partner_id` here.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: sitewit/models.py [hunk 4]
```diff
@@ -60,7 +60,8 @@ def create(cls, user, site_id, url, user_token=None):
         email = cls._get_email(user.id)
         user_name = cls._get_valid_user_name(user.name)
         result = cls.get_service().create_account(
-            site_id, url, user_name, email, 'USD', 'US', user_token)
+            site_id, url, user_name, email, 'USD', 'US', user_token,
+            user.partner_id)
```
Comment (RayeN) [inline sitewit/models.py]: I think we shouldn't pass partner_id if it's non-WL user

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,8 +1,8 @@
 # [Changelog](https://github.com/yola/sitewit/releases)
 
-## 0.8.1
+## 0.8.0
```
Comment (RayeN) [inline CHANGELOG.md]: DEV?

File: sitewit/services.py (hunk 3 above)
Comment (RayeN) [inline sitewit/services.py]: Since it's a service method - let's leave it as is.  Sorry.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: sitewit/services.py (hunk 3 above)
Comment (RayeN) [inline sitewit/services.py]: But we I think we should name it `remote_subpartner_id` as well, what do you think?

File: sitewit/services.py [hunk 6]
```diff
@@ -408,9 +429,11 @@ def create_partner(self, name, address, settings):
                 'address': address,
                 'whiteLabelSettings': settings}
 
+        if remote_id is not None:
+            data['remote_id'] = remote_id
```
Comment (RayeN) [inline sitewit/services.py]: `requests` do not send params with `None` values

File: sitewit/services.py (hunk 3 above)
Comment (stefanor) [inline sitewit/services.py]: Yeah, `remote_id` is a very confusing name.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: sitewit/services.py [hunk 7]
```diff
@@ -35,17 +35,34 @@ def _get_account_auth_header(self, account_token):
         return self._compose_auth_header((
             self._partner_id, self._partner_token, account_token))
 
-    def _get_partner_auth_header(self, subpartner_id=None):
+    def _get_partner_auth_headers(
+            self, subpartner_id=None, remote_subpartner_id=None):
+
+        if subpartner_id is not None and remote_subpartner_id is not None:
```
Comment (RayeN) [inline sitewit/services.py]: we could probably use `if subpartner_id and remote_subpartner_id:` here.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: sitewit/services.py [hunk 8]
```diff
@@ -398,19 +419,23 @@ def create_partner(self, name, address, settings):
...
             https://sandboxpapi.sitewit.com/Help/Api/GET-api-Partner
         """
-        data = {'name': name,
-                'address': address,
-                'whiteLabelSettings': settings}
+        data = {
+            'name': name,
+            'address': address,
+            'whiteLabelSettings': settings,
+            'remote_id': remote_id
```
Comment (stefanor) [inline sitewit/services.py]: I didn't see this parameter mentioned in the ticket or on the browseable API docs. Did I miss something?

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: sitewit/services.py (hunk 8 above)
Comment (ekoval) [inline sitewit/services.py]: Good catch, thanks @stefanor ! Fixed and corrected tests to detect this.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: sitewit/services.py (hunk 8 above)
Comment (stefanor) [inline sitewit/services.py]: I didn't see `remoteId` either

**Pull request merged**